### PR TITLE
✨ Support individual memo in nft-send

### PR DESCRIPTION
**Pull request by williamchong**

### CODE REVIEW

File: send-nft/index.js [hunk 1]
```diff
@@ -67,13 +68,8 @@ async function run() {
...
     const nftCountObject = data.reduce((object, item) => {
-      if (object[item.classId]) {
-        // eslint-disable-next-line no-param-reassign
-        object[item.classId] += 1;
-      } else {
-        // eslint-disable-next-line no-param-reassign
-        object[item.classId] = 1;
-      }
+      // eslint-disable-next-line no-param-reassign
+      object[item.classId] = (object[item.classId] || 1) + 1;
```
Comment (WeiJun0827) [inline send-nft/index.js:72]: Should be `0`?
```suggestion
      object[item.classId] = (object[item.classId] || 0) + 1;
```

File: send-nft/index.js [hunk 2]
```diff
@@ -108,36 +104,65 @@ async function run() {
...
+          {
+            amount: [
+              {
+                denom: DENOM,
+                amount: `${new BigNumber(1).shiftedBy(amountNeedDigits).toFixed(0)}`,
+              },
+            ],
+            gas: `${new BigNumber(1).shiftedBy(gasNeedDigits).toFixed(0)}`,
+          },
+          MEMO,
```
Comment (WeiJun0827) [inline send-nft/index.js:126]: Should use the memo from csv?
```suggestion
          data[i].memo || MEMO,
```

File: send-nft/index.js [hunk 3]
```diff
@@ -108,36 +104,65 @@ async function run() {
...
+        );
+        try {
+          await client.broadcastTx(tx, 1000, 1000);
```
Comment (WeiJun0827) [inline send-nft/index.js:129]: Do we expect to sign tx separately if any of the rows in .csv has the memo field?

File: send-nft/index.js [hunk 4]
```diff
@@ -108,36 +104,65 @@ async function run() {
...
-          denom: DENOM,
-          amount: `${new BigNumber(msgAnyArray.length).shiftedBy(amountNeedDigits).toFixed(0)}`,
+          amount: [
+            {
+              denom: DENOM,
+              amount: `${new BigNumber(data.length).shiftedBy(amountNeedDigits).toFixed(0)}`,
+            },
+          ],
+          gas: `${new BigNumber(data.length).shiftedBy(gasNeedDigits).toFixed(0)}`,
         },
```
Comment (WeiJun0827) [inline send-nft/index.js:160]: Maybe refactor this part and the above part to a function?

File: send-nft/index.js (hunk 3 above)
Comment (williamchong) [reply send-nft/index.js:129]: we would need 1 tx per memo, so yes

Comment (WeiJun0827) [reply send-nft/index.js:133]: Oh, I see

Comment (WeiJun0827) [reply send-nft/index.js:133]: Oh, I see. 
BTW, do we prefer to push those messages that don't have a memo to the array and sign them at once, or it's OK to keep them signed and broadcasted one by one?

Comment (williamchong) [reply send-nft/index.js:133]: signing doesnt takes much time except fetching sequence, so they should be the same
on second thought, you reminded me that i should increment and assign sequence manually ...

File: send-nft/index.js [hunk 5]
```diff
@@ -108,36 +117,66 @@ async function run() {
...
+            throw err;
+          }
+        }
+      } else {
+        const msgAny = formatMsgSend(
+          firstAccount.address,
+          e.address,
+          e.classId,
+          removed[0].id,
+        );
```
Comment (WeiJun0827) [inline send-nft/index.js:165]: Can move `msgAny` declaration out of `if...else...` and use it in both cases?

**Review by WeiJun0827 — APPROVED** [review]

**Pull request merged**